File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& arr) {
        int n=arr.size(),l=0,r=n-1,area=0,ans=0;//very important question
        while(l<r){
            int len=min(arr[l],arr[r]);
             area=max(len*(r-l),area);
            while(arr[r]<=len&&l<r)r--;
            while(arr[l]<=len&&l<r)l++;
        }
        return area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
        l,r=0,len(nums)-1
        res=0
        while(l<r):
            res=max(res,min(nums[l],nums[r])*(r-l))
            if nums[l]<=nums[r]:
                l+=1
            elif nums[r]<nums[l]:
                r-=1
        return res
        